spec/hooks_spec.rb

```diff
@@ -0,0 +1,129 @@
+# require File.expand_path 'spec_helper.rb', __dir__
+require 'ostruct'
+require 'byebug'
+
+RSpec.describe F1SalesCustom::Hooks::Lead do
+  describe '#switch_source' do
+    let(:lead) do
+      lead = OpenStruct.new
+      lead.source = source
+      lead.product = product
+      lead.customer = customer
+      lead.message = 'message'
+      lead.description = 'description'
+      lead.id = '123leadid'
+
+      lead
+    end
+
+    let(:source) do
+      source = OpenStruct.new
+      source.name = 'Other source'
+
+      source
+    end
+
+    let(:product) do
+      product = OpenStruct.new
+      product.name = 'product name'
+
+      product
+    end
+
+    let(:customer) do
+      customer = OpenStruct.new
+      customer.name = 'customer name'
+      customer.email = 'customer email'
+      customer.phone = 'customer phone'
+
+      customer
+    end
+
+    let(:switch_source) { described_class.switch_source(lead) }
+
+    context 'when lead is from Facebook' do
+      let(:call_url) { 'https://toribaveiculos.f1sales.org/public/api/v1/leads' }
+      let(:lead_class_double) { class_double('Lead').as_stubbed_const }
+      let(:count_lead) { 0 }
+      let(:lead_id) { '123leadid' }
+      let(:lead_created_payload) do
+        {
+          'data' => {
+            'id' => 'newleadabc123'
+          }
+        }
+      end
+
+      before do
+        source.name = 'Facebook'
+        stub_request(:post, call_url).with(body: lead_payload.to_json).to_return(status: 200, body: lead_created_payload.to_json, headers: {})
+        allow(lead_class_double).to receive(:where).with(source: source)
+                                                   .and_return(double('relation', count: count_lead))
+      end
+
+      let(:lead_payload) do
+        {
+          lead: {
+            message: lead.message,
+            description: lead.description,
+            customer: {
+              name: customer.name,
+              email: customer.email,
+              phone: customer.phone
+            },
+            product: {
+              name: product.name
+            },
+            transferred_path: {
+              from: 'toribapompeia',
+              id: lead_id
+            },
+            source: {
+              name: 'Facebook - Toriba Veículos Volkswagen'
+            }
+          }
+        }
+      end
+
+      context 'when count lead is even' do
```

Comment:
This test isn't pass. 

```
Failure/Error: response = HTTP.post('https://toribaveiculos.f1sales.org/public/api/v1/leads', json: lead_payload)
     
     WebMock::NetConnectNotAllowedError:
       Real HTTP connections are disabled. Unregistered request: POST https://toribaveiculos.f1sales.org/public/api/v1/leads with body '{"lead":{"customer":{"name":"customer name","email":"customer email","phone":"customer phone"},"product":{"name":"product name"},"source":{"name":"Facebook - Toriba Veículos Volkswagen"},"message":"message","description":"description","transferred_path":{"from":"toribapompeia","id":"123leadid"}}}' with headers {'Connection'=>'close', 'Content-Type'=>'application/json; charset=UTF-8', 'Host'=>'toribaveiculos.f1sales.org', 'User-Agent'=>'http.rb/5.1.1'}
     
       You can stub this request with the following snippet:
     
       stub_request(:post, "https://toribaveiculos.f1sales.org/public/api/v1/leads").
         with(
           body: "{\"lead\":{\"customer\":{\"name\":\"customer name\",\"email\":\"customer email\",\"phone\":\"customer phone\"},\"product\":{\"name\":\"product name\"},\"source\":{\"name\":\"Facebook - Toriba Veículos Volkswagen\"},\"message\":\"message\",\"description\":\"description\",\"transferred_path\":{\"from\":\"toribapompeia\",\"id\":\"123leadid\"}}}",
           headers: {
          'Connection'=>'close',
          'Content-Type'=>'application/json; charset=UTF-8',
          'Host'=>'toribaveiculos.f1sales.org',
          'User-Agent'=>'http.rb/5.1.1'
           }).
         to_return(status: 200, body: "", headers: {})
     
       registered request stubs:
     
       stub_request(:post, "https://toribaveiculos.f1sales.org/public/api/v1/leads").
         with(
           body: "{\"lead\":{\"message\":\"message\",\"description\":\"description\",\"customer\":{\"name\":\"customer name\",\"email\":\"customer email\",\"phone\":\"customer phone\"},\"product\":{\"name\":\"product name\"},\"transferred_path\":{\"from\":\"toribapompeia\",\"id\":\"123leadid\"},\"source\":{\"name\":\"Facebook - Toriba Veículos Volkswagen\"}}}")
     
       ===========
```

Reply:
I suspect is because the body is wrong: 

the string: `"{\"lead\":{\"message\":\"message\`
is different than the string: `"{\"lead\":{\"customer\"`

In this case the order of the hash keys matters because we are comparing string not hashes